functions/Get-DbaAgentJobOutputFile.ps1

```diff
@@ -0,0 +1,151 @@
+    This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License along with this program. If not, see http://www.gnu.org/licenses/.
+
+#>
+[CmdletBinding()]
+param
+(# The Server/instance 
+        [Parameter(Mandatory=$true,HelpMessage='The SQL Server Instance', 
+                   ValueFromPipeline=$true,
+                   ValueFromPipelineByPropertyName=$true, 
+                   ValueFromRemainingArguments=$false, 
+                   Position=0)]
+        [ValidateNotNull()]
+        [ValidateNotNullOrEmpty()]
+        [object]$SqlServer,
+        [Parameter(Mandatory=$false,HelpMessage='SQL Credential', 
+                   ValueFromPipeline=$true,
+                   ValueFromPipelineByPropertyName=$true, 
+                   ValueFromRemainingArguments=$false, 
+                   Position=1)]
+        [System.Management.Automation.PSCredential]$SqlCredential
+)
+    DynamicParam { if ($SqlServer) { return (Get-ParamSqlJobs -SqlServer $SqlServer -SqlCredential $SourceSqlCredential) } }
+
+	BEGIN 
+    {
+		$jobname = $psboundparameters.Jobs   
+    }
+    PROCESS
+    {
+        foreach ($instance in $sqlserver)
+        {
+            $Server = Connect-SqlServer -SqlServer $instance  -SqlCredential $SqlCredential
+            $jobs = $Server.JobServer.Jobs
+            if ($JobName)
+            {
+                try
+                {
+                    $Jobs = $server.JobServer.Jobs[$JobName]
+                }
+                catch
+                {
+                    Write-Warning "Cannot find $JobName on $SQLServer"
+                }
+            }
+
+            foreach($Job in $Jobs)
+            {
+                foreach($Step in $Job.JobSteps)
+                {
+                    if($Step.OutputFileName)
+                    {
+                        $fileName = Join-AdminUNC $Server.ComputerNamePhysicalNetBIOS $Step.OutputFileName
+                        [pscustomobject]@{
+                        ComputerName = $Server.ComputerNamePhysicalNetBIOS
+                        InstanceName = $Server.Instancename
+                        SqlInstance = $Server.Name
+                        Job = $Job.Name
+                        JobStep = $step.Name
+                        OutputFile = $FileName
+                        }
+                    }
+                    else
+                    {
+                        Write-Verbose "$(($Step).Name) for $JobName has No Output File"
+                    }
+                }
+            }      
+         
+            $server.ConnectionContext.Disconnect()
+        }
+    }
+    END
```

Comment:
I like END {}
but the majority won't have it, so we have to remove it everywhere 😞